### PR TITLE
PHPM_51

### PR DESCRIPTION
**Pull request by hetvi-7span**
As a user of our platform, I want to be able to retrieve a list of friend requests that have been sent to me so that I can manage my social connections more efficiently.

Acceptance Criteria:

API endpoint is created with the following endpoint: GET /friend-requests

The API endpoint returns a list of friend requests in JSON format

The list includes the name, username, and profile picture of the users who sent the friend request

The list includes the date and time the friend request was sent

The API endpoint handles errors gracefully and returns appropriate error messages when necessary

Additional Considerations:

Pagination is implemented for the API endpoint in case the list of friend requests is too long to return in a single response

Authentication is implemented for the API endpoint to ensure that only authorized users can retrieve the list of friend requests

### CODE REVIEW

File: friendlist/src/main/java/com/bytro/friendlist/rest/controller/FriendRequestController.java [hunk 1]
```diff
@@ -31,4 +35,10 @@ public BaseResponse<Void> acceptRejectFriendRequest(
             @RequestBody AcceptRejectFriendRequest acceptRejectFriendRequest) {
         return friendRequestHandler.acceptRejectFriendRequest(acceptRejectFriendRequest);
     }
+    @Operation(summary = "List of friend request")
+    @GetMapping("/friend-requests")
```
Comment (akshay-7span) [inline friendlist/src/main/java/com/bytro/friendlist/rest/controller/FriendRequestController.java:39]: we can take this userId in the path variable and page and size in request param

**Review by akshay-7span — CHANGES_REQUESTED** [review]
Add logs so that we can track request

File: friendlist/src/main/java/com/bytro/friendlist/rest/controller/FriendRequestController.java (hunk 1 above)
Comment (hetvi-7span) [reply friendlist/src/main/java/com/bytro/friendlist/rest/controller/FriendRequestController.java:39]: at the end, we will take user-id from the token so as of now we can continue with this

**Review by akshay-7span — APPROVED** [review]

**Comment by hetvi-7span** [comment]
patch-4

**Pull request merged**